include/benchmark/benchmark.h

```diff
@@ -685,8 +688,21 @@ class State {
   void FinishKeepRunning();
   internal::ThreadTimer* timer_;
   internal::ThreadManager* manager_;
+  int num_thread_states_;
 
   friend struct internal::BenchmarkInstance;
+  friend class ThreadState;
+};
+
+class ThreadState : public State
+{
```

Comment:
note we use google style so this (and other braces) should be on the previous line. you can use clang-format to make this easier.